java/com/engflow/binaryinput/README.md

```diff
@@ -0,0 +1,52 @@
+# Multiple Binary Input Example
+
+## Overview
+
+The goal of this example project is to test the performance of Engflow's remote execution and caching service based on the number of input binary files in the dependency graph. The project contains a `genrule` that generates a specified number of Java binaries for the `genbinary` Java library, which are then listed as dependencies in the main binary. The `Main.java` file loops through each generated class and calls its `greetNum` method.
+
+## Project Structure
+
+- `java/com/engflow/binaryinput/Main.java`: Main class that dynamically loads and invokes methods from generated classes.
+- `java/com/engflow/binaryinput/BUILD`: Bazel build file for the `main` java binary and the `genbinary` library.
+
+## Usage
+
+To generate the test files, build the `genbinary` library using the `genrule`:
+```sh
+bazel build //java/com/engflow/binaryinput:genbinary
+```
+
+Then, the program can be run with the following command:
+```sh
+bazel run //java/com/engflow/binaryinput:main
+```
+
+## How It Works
+
+1. **Generation of Java Binaries:**
+    - The `genrule` in the `BUILD` file generates a specified number of Java classes (`Hello1.java`, `Hello2.java`, ..., `HelloN.java`).
+    - Each generated class contains a `greetNum` method that prints a unique message.
+
+2. **Main Class Execution:**
+    - The `Main.java` file in `binaryinput` dynamically loads each generated class using reflection.
```

Comment:
Very nice. Please add more on why you use this (the classes names are created dynamically too.